.github/workflows/build-bundle.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: Build Bundle
+
+on:
+  pull_request:
+    branches:
+    - KU-689/publish-bundle
+  push:
+
+jobs:
+  build-charms:
+    name: Build and push bundles
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Pack Bundle
+        run: |
+          sudo snap install charmcraft --classic --channel latest/stable
+          pushd main
+          charmcraft pack -v 
+          echo "BUNDLE_FILE=$(ls ./*.zip)" >> $GITHUB_ENV
+      
+      - name: Upload bundle to edge
+        uses: canonical/charming-actions/upload-charm@2.5.0-rc
+        with:
+          charm-path: main
+          built-charm-path: ${{ env.BUNDLE_FILE }}
```

Comment:
🪄  yep, this uses the upload-charm action with a specific charm-path to the bundle 

there is an upload-bundle action but it's broken